src/mito2/src/worker/handle_create.rs

```diff
@@ -55,13 +55,21 @@ impl<S: LogStore> RegionWorkerLoop<S> {
         }
         builder.primary_key(request.primary_key);
         let metadata = builder.build().context(InvalidMetadataSnafu)?;
-
+        let object_store = if let Some(custom_storage) = request.options.get("storage") {
+            self.object_store_manager
+                .find(custom_storage)
+                .context(ObjectStoreNotFoundSnafu {
+                    object_store: custom_storage.to_string(),
+                })?
+        } else {
+            self.object_store_manager.default_object_store()
+        };
```

Comment:
suggestion: Can we pass the object store manager to the `RegionOpener`? Then we can add a `storage` field to the `RegionOptions` and choose the object store for the region in `RegionOpener::create_or_open()`.